app/controllers/posts_controller.rb

```diff
@@ -7,5 +7,29 @@ def index
   def show
     @post = Post.find(params[:id])
     @comments = @post.comments
+    @likes = @post.likes
+  end
+
+  def new
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text, :author))
+    @post.author = current_user
+    respond_to do |format|
+      format.html do
+        if @post.save
+          flash[:success] = 'Post saved successfully'
+          redirect_to users_url
+        else
+          flash.now[:error] = 'Error: Post could not be saved'
+          render :new, locals: { post: @post }
+        end
+      end
+    end
   end
```

Comment:
- [ **OPTIONAL** ] **RECOMMENDED:** I think it would be better if you use the **"[the strong parameters](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)"** like you studied in **[this lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)** for the project requirements